### DISCUSSION TITLE
raspberry ignition code

### DISCUSSION THREAD

**Discussion by mforest01** [PiSupply/PiJuice#823] [Q&A]
Hello, as part of a project, I'm trying to understand how pijuice works. Therefore, I am looking but cannot find the code that is executed when the ignition button is pressed. Would it be possible to help me on this point? Thanks in advance.

**Comment by tvoverbeek** [comment]
What do you mean by ignition code?
Press of SW1?

**Reply by mforest01** [reply]
I mean the program that allows you to wake up the raspberry

**Reply by tvoverbeek** [reply]
That is the firmware for the STM32 MCU on the PiJuice board.
Firmware Source can be found on https://github.com/PiSupply/PiJuice/tree/master/Firmware

**Reply by mforest01** [reply]
Thank you. In fact, my goal is to ensure that the pijuice can turn off the raspberry and turn it back on 30 seconds later if one of the I2C sensors detects an infrared.

**Reply by tvoverbeek** [reply]
Look at the wakeup possibility via the IO2 pin of the PiJuice:
https://github.com/PiSupply/PiJuice/tree/master/Software#pijuice-hat-config-io-menu
Suggest to connect your  infrared sensor to this pin.
No way (as far as I know) to do this when the infrared sensor is connected to the I2C bus of the RPi.

**Reply by mforest01** [reply]
It's not really the sensors that allow the restart, in fact, in case of infrared detection, I would like the raspberry to be automatically turned off and then restart 30 seconds later

**Reply by tvoverbeek** [reply]
Ok, then the other way around. Detect the infrared via the Pi and on detection program the PiJuice alarm to wake up the Pi 40 sec later then shutdown the Pi (assuming the shutdown itself takes ca 10 sec)

**Reply by mforest01** [reply]
Okay thank you so much. In this case, is it possible to program an alarm automatically?

**Reply by mforest01** [reply]
In fact, that's where I'm stuck.

**Reply by tvoverbeek** [reply]
I suppose you are polling your IR sensors. When you detect something then start a python script which reads the current time and then programs the alarm 40 sec  from the current time ans at the end of the script shut down the Pi.
Note the PiJuice RTC alarm uses UTC, so you have to convert from local time to UTC.
See https://github.com/PiSupply/PiJuice/tree/master/Software#pijuice-rtc-alarm.

**Reply by mforest01** [reply]
Thank you very much for your help!
